slack-base/src/main/java/com/hubspot/slack/client/models/interaction/json/BlockElementActionDeserializer.java

```diff
@@ -0,0 +1,42 @@
+package com.hubspot.slack.client.models.interaction.json;
+
+import java.io.IOException;
+import java.util.Optional;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.ObjectCodec;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import com.hubspot.slack.client.models.blocks.elements.BlockElement;
+import com.hubspot.slack.client.models.interaction.BlockElementAction;
+
+public class BlockElementActionDeserializer extends StdDeserializer<BlockElementAction> {
```

Comment:
Just a quick q: can you comment on why we needed to go the explicit deserializer route rather than straight jackson?

Reply:
The problem is that you can't deserialise a polymorphic type with `@JsonUnwrapped` using Jackson. Have a look at [this stackoverflow question](https://stackoverflow.com/questions/37423848/deserializing-polymorphic-types-with-jsonunwrapped-using-jackson) for a summary

There was a bug raised about this against jackson years ago, but was closed as [will not fix](https://markmail.org/message/cyeyc2ousjp72lh3)

There's a similar issue where the maintainers also said it would not be supported. https://github.com/FasterXML/jackson-databind/issues/81

I spent a ridiculous amount of time on this (since I _hate_ having to write a customer serialiser/deserialiser), but I couldn't figure out any other way to achieve t.

Reply:
👍 thanks for the explanation